src/kem/sike/external/fpx.c

```diff
@@ -280,15 +280,15 @@ __inline unsigned int mp_sub(const digit_t* a, const digit_t* b, digit_t* c, con
 
 __inline static void mp_subaddfast(const digit_t* a, const digit_t* b, digit_t* c)
 { // Multiprecision subtraction followed by addition with p*2^MAXBITS_FIELD, c = a-b+(p*2^MAXBITS_FIELD) if a-b < 0, otherwise c=a-b. 
-#if !defined(USE_SIKE_ASM)
+#if (OS_TARGET == OS_WIN) || defined(GENERIC_IMPLEMENTATION) || (TARGET == TARGET_ARM)
     felm_t t1;
 
     digit_t mask = 0 - (digit_t)mp_sub(a, b, c, 2*NWORDS_FIELD);
     for (int i = 0; i < NWORDS_FIELD; i++)
         t1[i] = ((digit_t*)PRIME)[i] & mask;
     mp_addfast((digit_t*)&c[NWORDS_FIELD], t1, (digit_t*)&c[NWORDS_FIELD]);
 
-#else
+#elif (OS_TARGET == OS_NIX)               
```

Comment:
The P434 and P610 parameter sets only have x86-64 assembly versions of mp_subaddx2 if the CPU supports BMI2 instructions, so this check is not sufficient. There are similar problems for the conditions that wrap calls to mp_dblsub610x2, mp_add610, and the 434 versions of these.

Reply:
Ah yes, my last update overwrote correct checks for this; I'll revert them back.